ci_scripts/utils/git_helpers.py

```diff
@@ -0,0 +1,122 @@
+        return new.format(GH_TOKEN=os.getenv(ENVVAR_GIT_TOKEN))
+
+    def checkout(self, branch):
+        subprocess.check_call(['git', 'checkout', '-f', branch])
+
+    def add(self, path):
+        if isinstance(path, list):
+            for element in path:
+                self.add(element)
+        else:
+            subprocess.check_call(['git', 'add', path])
+
+    def commit(self, message):
+        subprocess.check_call(['git', 'commit', '-m', message])
+
+    def get_current_branch(self):
+        return subprocess.check_output(
+            ['git', 'rev-parse', '--abbrev-ref', 'HEAD']).decode(
+            'utf-8').strip()
+
+    def get_commit_count(self):
+        return subprocess.check_output(
+            ['git', 'rev-list', '--count', 'HEAD']).decode('utf-8').strip()
+
+    def get_commit_hash(self):
+        return subprocess.check_output(['git', 'rev-parse', 'HEAD']).decode(
+            'utf-8').strip()
+
+    def get_branch_point(self, commit1, commit2):
+        return subprocess.check_output(
+            ['git', 'merge-base', commit1, commit2]).decode('utf-8').strip()
+
+    def merge(self, branch):
+        subprocess.check_call(
+            ['git', 'merge', '--no-commit', '--strategy-option', 'theirs',
+             branch])
+        self.commit('Merge from %s' % branch)
+
+    def get_remote_url(self):
+        return subprocess.check_output(
+            ['git', 'remote', 'get-url', REMOTE_ALIAS]).decode('utf-8').strip()
+
+    def cherry_pick(self, commit_hash):
+        subprocess.check_call(['git', 'cherry-pick', commit_hash])
+
+    def set_remote_url(self, url):
+        subprocess.check_call(['git', 'remote', 'set-url', REMOTE_ALIAS, url])
+
+    def get_remote_branch(self, branch):
+        return '%s/%s' % (REMOTE_ALIAS, branch)
+
+    def set_upstream_branch(self, branch):
+        subprocess.check_call(
+            ['git', 'branch', '--set-upstream-to',
+             self.get_remote_branch(branch)])
+
+    # the output of the git command looks like:
+    # 'A       docs/news/789.feature'
+    def get_changes(self, commit1, commit2, dir=None):
+        diff_command = ['git', 'diff', '%s...%s' % (commit1, commit2),
+                        '--name-status']
+        if dir:
+            diff_command.append(dir)
+        return subprocess.check_output(diff_command).decode('utf-8')
+
+    def delete_branch(self, branch):
+        subprocess.check_call(['git', 'branch', '-D', branch])
+
+    def branch_exists(self, branch):
+        branches = subprocess.check_output(['git', 'branch']).decode(
+            'utf-8').strip()
+        return len([line for line in branches.splitlines() if
+                    line.replace('*', '').split() == branch.split()]) > 0
+
+    def get_changes_list(self, change_type, commit1, commit2, dir=None):
+        file_diff = self.get_changes(commit1, commit2, dir)
+        return [(line.strip()[len(change_type):]).strip() for line in

```

Comment:
I could do with some help here as to what is being done.

Reply:
part of #7